### ISSUE TITLE
Feature request: Adding stack functionality to MonadTall / Adding insert to different stack to Columnsstack

### ISSUE THREAD

**Issue by mcranium** [qtile/qtile#2321]
# Feature description
I am a huge fan of the Columns layout but for some tasks it is not really efficient. For example if you have a file manager on one side and a stack of pdfs on the other side. Moving through the stack is very efficient but every time you open up a new pdf you have to move it to the other stack. With the MonadTall layout this is not needed but there you don't have the ability to efficiently cycle through a stack of windows that occupy the full height of the screen.
To make such tasks more efficient I see two ways:

* Add the split stack functionality of the Columns layout to MonadTall

* Add new options in the Columns layout about where to insert new windows. Currently the options are "0" and "1" for above or below the active window in the current stack. This could be turned into "left", "right", "up" and "down"

Unfortunately, after going through the source code I realized that my Python skills are probably too limited to get this done on my own, but I am willing to help wherever I can.

# Qtile version

0.17.0

**Comment by Fabien-Bernard** [comment]
Maybe your workflow could be achieved with the Slice layout : 
- "match" key corresponding to your file manager,
- "fallback" key correponding to the VerticalTile layout for your pdf stack.

It's just a suggestion, doesn't mean anything about the feature proposition :)

**Comment by mcranium** [comment]
I found out that my desired functionality can be achieved in XMonad by using the TwoPane layout and the option to insert windows in the stack below the currently focussed window. 
I also had another idea how one could achieve this in qtile:
In the Column layout one could add a function like this: 
```
if in the left stack the number of windows gets larger than 1:
    take the lowermost window of the left stack and move it to the right stack
```
Then one could make this function optional through the config file (something like `single_master = True`).

I tried the approach with the Slice layout, but as soon as I open a window that matches the key qtile crashes. I saw that someone already made a bug report for this.

**Comment by Fabien-Bernard** [comment]
I do not experience crash.
Using thunar as file manager, this is the line I added to my config : 
`
layout.Slice(width=800, match=Match(wm_class=["thunar"]), fallback=layout.VerticalTile()),
`

I use the git-version of qtile, not the v0.17, so maybe it's why. But without more details, it's difficult to be sure. Which issue are you refering to?

**Comment by mcranium** [comment]
I just checked it again and the slice layout works just as it should. Maybe the issue has been fixed or I did something  wrong the last time, I am sorry for the confusion. 
Now that it works with the slice layout, there is just one thing missing to achieve what I was looking for. I noticed that my window resize commands are of course only working for the fallback layout. Is there a way to dynamically change the width of the slice window, so that I can resize it using keyboard shortcuts?
Many thanks for helping me out here!
